src/library/f3dOptions.h

```diff
@@ -0,0 +1,60 @@
+#ifndef f3dOptions_h
+#define f3dOptions_h
+
+#include <map>
+#include <string>
+#include <sstream>
+#include <iterator>
+#include <vector>
+#include <iostream>
+
+// TODO put somewhere
```

Comment:
So how do I hide this ?

I need to access it from the template code, which must be visible by the includer.
Not a template exert, I may be missing something

@CharlesGueunet @Meakk

Reply:
Put the implementation of `get` and `set` in the cxx file and declare all the accepted specializations in the cxx file too.  
The `operator>>` and `operator<<` can be declared in a private header included in `f3dOptions.cxx`

Reply:
since set/get are template, should the implementation not be in the header ?

Reply:
You don't have to, as long as you are explicitly specializing the supported types in the cxx.

Reply:
but why would I want to specialize templates ?

Reply:
found the right way, I think.